### ISSUE TITLE
Integrate with SOS by using Iceland

### ISSUE THREAD

**Issue by nuest** [52North/OpenSensorSearch#49]

**Comment by nuest** [comment]
@autermann @CarstenHollmann Can you confirm that, if we switch OpenSensorSearch to be based on iceland, it should be very straightforward to add it to the (existing or a new one) webapp-bundle? Then configuration and endpoints etc. should "just work", correct?

**Comment by nuest** [comment]
@autermann @CarstenHollmann Would it be possible to "add" example requests to the test client from a different file? I don't want to overwrite the example requests in the bundle, but include an additional file with example requests to be shown (given the respective endpoints are activated) and so forth.

**Comment by autermann** [comment]
Currently that's not possible. We're loading the example requests from a single JSON file called `client-config.json` via JavaScript, so we can't just load a set of file e.g. via a file pattern.

But, as that file is already 5k lines long, it would make sense to allow splitting it up... Something like this should be viable:

```json
{
  "examples": [
    "static/conf/examples/sos_v100.json",
    "static/conf/examples/sos_v20.json",
    "static/conf/examples/aqd_v10.json",
  ]
}
```

Then you still have to replace the file, but you have to provide far less content...

**Comment by nuest** [comment]
Or why not reading all files from a directory instead of having this "index" file? Then an overlay could just add a file, or also replace the index file if they do not want other examples to be loaded.

```
{
  "examples": [
    "static/conf/examples/",
    "static/conf/examples/*.json",
  ]
}
```

But once the examples are loaded, then the filtering within the client works, right?

I also assume that iceland does not include the test client - correct?

**Comment by autermann** [comment]
Am 15.06.2015 18:29 schrieb "Daniel Nüst" <notifications@github.com>:
>
> Or why not reading all files from a directory instead of having this
"index" file? Then an overlay could just add a file, or also replace the
index file if they do not want other examples to be loaded.
>
> {
>   "examples": [
>     "static/conf/examples/",
>     "static/conf/examples/*.json",
>   ]
> }
>

And how do I load all files in a directory from a browser using JavaScript?

> But once the examples are loaded, then the filtering within the client
works, right?
>

It should :-)

> I also assume that iceland does not include the test client - correct?
>

That's right...

**Comment by nuest** [comment]
Good point about the javascript loading :-) Regexes also won't work. Ok - I updated the first entry to reflect the agreement on example requests.

**Comment by nuest** [comment]
Will work on this within https://github.com/nuest/OpenSensorSearch/tree/feature/iceland
